analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CheckArgumentException.CSharp12.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+// Reproducer for https://github.com/SonarSource/sonar-dotnet/issues/7714
+public class PrimaryConstructor(string s, string q)
+{
+    private readonly string _s = s ?? throw new ArgumentNullException(nameof(s)); // Compliant: s is a class parameter
```

Comment:
Do we support `ArgumentNullException.ThrowIfNull(s)` with this rule? If so, we should add a test case with `ArgumentNullException.ThrowIfNull(s, nameof(q))`.

https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception.throwifnull?view=net-7.0

Reply:
No we don't I added a FN reproducer for that